lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -247,6 +247,7 @@ def commit(action, parameters, option)
           message_from(action, response),
           response,
           authorization: authorization_from(action, response),
+          avs_result: AVSResult.new(code: response['result'].first&.dig('transaction', 'avs', 'result')),
```

Comment:
Small suggestion: extract out the AVS parsing logic into an `avs_from` method and guard against possibility of no AVS result returned?

Reply:
Done!